src/main/flight/mixer.c

```diff
@@ -743,27 +748,125 @@ STATIC_UNIT_TESTED void servoMixer(void)
-            axisPID[ROLL] * currentMixer[i].roll +
-            -mixerConfig->yaw_motor_direction * axisPID[YAW] * currentMixer[i].yaw;
-    }
+    if (IS_RC_MODE_ACTIVE(BOXAIRMODE)) {
+        // Initial mixer concept by bdoiron74 reused and optimized for Air Mode
+        int16_t rollPitchYawMix[MAX_SUPPORTED_MOTORS];
+        int16_t rollPitchYawMixMax = 0; // assumption: symetrical about zero.
+        int16_t rollPitchYawMixMin = 0;
+
+        // Find roll/pitch/yaw desired output
+        for (i = 0; i < motorCount; i++) {
+            rollPitchYawMix[i] =
+                axisPID[PITCH] * currentMixer[i].pitch +
+                axisPID[ROLL] * currentMixer[i].roll +
+                -mixerConfig->yaw_motor_direction * axisPID[YAW] * currentMixer[i].yaw;
 
-    if (ARMING_FLAG(ARMED)) {
+            if (rollPitchYawMix[i] > rollPitchYawMixMax) rollPitchYawMixMax = rollPitchYawMix[i];
+            if (rollPitchYawMix[i] < rollPitchYawMixMin) rollPitchYawMixMin = rollPitchYawMix[i];
+        }
+
+        // Scale roll/pitch/yaw uniformly to fit within throttle range
+        int16_t rollPitchYawMixRange = rollPitchYawMixMax - rollPitchYawMixMin;
+        int16_t throttleRange, throttle;
+        int16_t throttleMin, throttleMax;
+        throttle3dState_e throttle3dPosition = THROTTLE_3D_NEUTRAL;  // Initiate in Neutral before all throttle checks
+        throttle = rcCommand[THROTTLE];
+
+        // Find min and max throttle based on condition
+        if (feature(FEATURE_3D)) {
+            static int16_t throttleMinPrevious = 0, throttleMaxPrevious = 0, throttlePrevious = 0;
+
+            if (rcData[THROTTLE] <= (rxConfig->midrc - flight3DConfig->deadband3d_throttle)) {
+                throttleMax = flight3DConfig->deadband3d_low;
+                throttleMin = escAndServoConfig->minthrottle;
+                throttle3dPosition = THROTTLE_3D_NEGATIVE;
+            } else if (rcData[THROTTLE] >= (rxConfig->midrc + flight3DConfig->deadband3d_throttle)) {
+                throttleMax = escAndServoConfig->maxthrottle;
+                throttleMin = flight3DConfig->deadband3d_high;
+                throttle3dPosition = THROTTLE_3D_POSITIVE;
+            } else {
+                // when starting in neutral go positive and  when coming from positive. Keep positive throttle within deadband
+                if ((throttle3dPosition == THROTTLE_3D_NEUTRAL && !throttleMinPrevious)
+                    || (throttle3dPosition == THROTTLE_3D_NEUTRAL && (throttleMinPrevious >= flight3DConfig->deadband3d_high))) {
+                    throttleMax = escAndServoConfig->maxthrottle;
+                    throttle = throttleMin = flight3DConfig->deadband3d_high;
+                // When coming from negative. Keep negative throttle within deadband
+                } else if (throttle3dPosition == THROTTLE_3D_NEUTRAL && (throttleMinPrevious <= flight3DConfig->deadband3d_low)) {
+                    throttleMax = escAndServoConfig->mincommand;
```

Comment:
I didn't check that far - assignment  in THROTTLE_3D_NEGATIVE sets throttleMin < throttleMax, in this code it is other way.